### PR TITLE
Provide links to Cops' explanations if available

### PR DESCRIPTION
**Pull request by tomasz**
Sometimes the one-line explanation of the cop isn't enough and the examples provided over at [ruby-style-guide](https://github.com/bbatsov/ruby-style-guide) usually help clear things up. With these changes, if the Cop's name is specified and has an explanation then the Cop's name will be a clickable link to that explanation.

_P.S. While I think I've followed the contribution guidelines (`lint` and `test` pass), sorry in advance for any mistakes on my part._

### CODE REVIEW

File: lib/index.coffee [hunk 1]
```diff
@@ -30,9 +41,10 @@ lint = (editor) ->
     throw new Error stderr or stdout unless typeof parsed is 'object'
     (parsed.files?[0]?.offenses or []).map (offense) ->
       {cop_name, location, message, severity} = offense
+      {cop_name, message} = extractUri cop_name, message
```
Comment (Arcanemagus) [inline lib/index.coffee]: `message` isn't being modified in this function, there's no need to return it here.

File: lib/index.coffee [hunk 2]
```diff
@@ -30,9 +41,10 @@ lint = (editor) ->
...
       {line, column, length} = location or DEFAULT_LOCATION
       type: if WARNINGS.has(severity) then 'Warning' else 'Error'
-      text: (message or DEFAULT_MESSAGE) +
+      html: (message or DEFAULT_MESSAGE) +
```
Comment (Arcanemagus) [inline lib/index.coffee]: You need something to escape any HTML already in the message (`escape-html` package is used in other linters for this purpose). `extractUri` would be a good place to put this (and a reason for it to return `message` :wink:).

File: spec/linter-rubocop-spec.js [hunk 3]
```diff
@@ -64,13 +64,14 @@ describe('The RuboCop provider for Linter', () => {
...
-        'interpolation or special symbols. (Style/StringLiterals)';
+        'interpolation or special symbols. ' +
+        '(<a href="https://github.com/bbatsov/ruby-style-guide#consistent-string-literals">Style/StringLiterals</a>)';
 
       waitsForPromise(() =>
         lint(editor).then(messages => {
           expect(messages[0].type).toEqual('Warning');
-          expect(messages[0].html).not.toBeDefined();
-          expect(messages[0].text).toEqual(msgText);
+          expect(messages[0].html).toEqual(msgText);
```
Comment (Arcanemagus) [inline spec/linter-rubocop-spec.js]: `toEqual` (`==`) ->`toBe` (`===`)

File: spec/linter-rubocop-spec.js [hunk 4]
```diff
@@ -44,8 +44,8 @@ describe('The RuboCop provider for Linter', () => {
       waitsForPromise(() =>
         lint(editor).then(messages => {
           expect(messages[0].type).toEqual('Error');
-          expect(messages[0].html).not.toBeDefined();
-          expect(messages[0].text).toEqual(msgText);
+          expect(messages[0].html).toEqual(msgText);
```
Comment (Arcanemagus) [inline spec/linter-rubocop-spec.js]: `toEqual` (`==`) ->`toBe` (`===`)

**Comment by Arcanemagus** [comment]
Thanks for getting this started! It's been a great feature in the other linters that have added it so I'm looking forward to getting it out here for the users 😁.

**Comment by tomasz** [comment]
Hi! Sorry for the delay, but I'm back.

A couple of things changed, but this comment is still relevant:

> `message` isn't being modified in this function, there's no need to return it here.

It's may be modified. With the `--display-style-guide` flag active, Rubocop's messages look like this:

    Line is too long. [81/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)

The regexp in `extractUrl` function removes the `⎵(http://example.org/)` part from the original `message`, so it gets modified. If there is no style guide for the cop, then the `message` stays the same.

**Comment by Arcanemagus** [comment]
The CI build status is all green, merging!

**Pull request merged**